frontend/src/js/components/tenants/tenantlist.tsx

```diff
@@ -0,0 +1,177 @@
+};
+export const DeviceLimitRender = (props: RendererProp) => {
+  const deviceLimit = useSelector(getDeviceLimit);
+  const { column, tenant } = props;
+  const attributeValue = tenant[column.attribute.name];
+  return `${attributeValue}/${deviceLimit}`;
+};
+export const boolRender = (props: RendererProp) => {
+  const { column, tenant } = props;
+  return <div>{tenant[column.attribute.name] ? <CheckIcon /> : <div>-</div>}</div>;
+};
+const AttributeRenderer = ({ content, textContent }) => (
+  <div title={typeof textContent === 'string' ? textContent : ''}>
+    <div className="text-overflow">{content}</div>
+  </div>
+);
+const dateRender = (props: RendererProp) => {
+  const { column, tenant } = props;
+  const attributeValue = dayjs(tenant[column.attribute.name]).format('YYYY-MM-DD HH:mm');
+  return <AttributeRenderer content={attributeValue} textContent={tenant[column.attribute.name]}></AttributeRenderer>;
+};
+const moreDetailsRender = () => {
+  return (
+    <div className="link-color">
+      View details <ArrowForwardIcon sx={{ fontSize: 12 }} />
+    </div>
+  );
+};
+const columnHeaders: ColumnHeader[] = [
+  {
+    title: 'Name',
+    attribute: {
+      name: 'name',
+      scope: ''
+    },
+    sortable: false,
+    textRender: defaultTextRender
+  },
+  {
+    title: 'Devices',
+    attribute: {
+      name: 'device_limit',
+      scope: ''
+    },
+    sortable: false,
+    textRender: DeviceLimitRender
+  },
+  {
+    title: 'Delta updates enabled ',
+    attribute: {
+      name: 'binary_delta',
+      scope: ''
+    },
+    sortable: false,
+    textRender: boolRender
+  },
+  {
+    title: 'Created',
+    attribute: {
+      name: 'created_at',
+      scope: ''
+    },
+    sortable: false,
+    textRender: dateRender
+  },
+  {
+    title: 'More details',
+    attribute: {
+      name: '',
+      scope: ''
+    },
+    sortable: false,
+    textRender: moreDetailsRender
+  }
+];
+
+interface DefaultAttributeRendererProps {
+  column: ColumnHeader;
+  tenant: Tenant;
+}
+
+const DefaultAttributeRenderer = (props: DefaultAttributeRendererProps) => {
+  const { column, tenant } = props;
+  const text = column.textRender({ tenant, column });
+  return <AttributeRenderer content={text} textContent={text} />;
+};
+
+const TenantListItem = (props: ListItemComponentProps<Tenant>) => {
+  const { listItem, columnHeaders, onClick } = props;
+  const handleOnClick = useCallback(() => {
+    onClick(listItem);
+  }, [listItem.id, onClick]);
+
+  return (
+    <div onClick={handleOnClick} className={`deviceListRow deviceListItem clickable`}>
+      {columnHeaders.map((column: ColumnHeader) => {
+        const Component = column.component ? column.component : DefaultAttributeRenderer;
+        return <Component column={column} tenant={listItem} key={column.title} />;
+      })}
+    </div>
+  );
+};
+export const TenantList = ({ customColumnSizes, idAttribute }) => {
```

Comment:
the `idAttribute` is probably only relevant for the device list, similarly `customColumnSizes` as the device list potentially contains much more information than the `TenantList`